test_fms/time_interp/test_time_interp.F90

```diff
@@ -23,19 +23,24 @@ program test_time_interp
  use time_manager_mod, only: get_date, set_time, set_date, time_manager_init, set_calendar_type, operator(+)
  use time_manager_mod, only: JULIAN, time_type, increment_time, NOLEAP, print_date
  use  time_interp_mod, only: time_interp_init, time_interp, NONE, YEAR, MONTH, DAY
+ use time_manager_mod, only: operator(<=), operator(>=), operator(==)
+ use platform_mod
 
  implicit none
 
- integer, parameter :: num_Time=6
+ integer, parameter :: num_Time=6, kindl = TI_TEST_KIND_
  type(time_type) :: Time_beg, Time_end, Time(num_Time)
  type(time_type), allocatable, dimension(:) :: Timelist
- integer :: index1, index2, mo, yr, timelist_len, outunit, ntest, nline
- real :: weight
+ integer :: index1, index2, mo, yr, outunit, ntest, nline
+ real(TI_TEST_KIND_) :: weight
+ real(TI_TEST_KIND_) :: ref_weights(num_Time), ref_weights_leap(num_Time)
+ real(TI_TEST_KIND_), parameter :: SMALL = 1.0e-7_kindl ! r4 will fail with 8
+ real(TI_TEST_KIND_), parameter :: midpoint = 0.483870967741935_kindl
+ real(TI_TEST_KIND_), parameter :: day_before_leap_day =  0.964285714285714_kindl
```

Comment:
where did these numbers come from?

Reply:
Just reference weights from the calculation, its a fraction of time passed over a interval. So for this its the midway point of a month (jan 16). I think i commented on it a bit lower down.

I might be able to replace these with the actual time_type calculation if thats preferred, i guess i just like seeing what the actual value ends up as.

Reply:
could the actual time_type calculation be used and add the numbers as comments?  Is this perhaps the reason you're not getting exactly agreeing answers?

Reply:
After looking into it i don't think i can reproduce the calculation unless i copy in a whole routine (set_modtime). I think this way has some benefits though, like if the time_type arithmentic overrides return the wrong thing a check with the same operations will just be checking wrong answers against wrong answers.